### PR TITLE
Add checksum for py-ipywidgets@7.7.0

### PR DESCRIPTION
**Pull request by aandvalenzuela**

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-ipywidgets/package.py [hunk 1]
```diff
@@ -12,6 +12,7 @@ class PyIpywidgets(PythonPackage):
     homepage = "https://github.com/ipython/ipywidgets"
     pypi = "ipywidgets/ipywidgets-7.6.5.tar.gz"
 
+    version('7.7.0', sha256='ab4a5596855a88b83761921c768707d65e5847068139bc1729ddfe834703542a')
     version('7.6.5', sha256='00974f7cb4d5f8d494c19810fedb9fa9b64bffd3cda7c2be23c133a1ad3c99c5')
     version('7.6.3', sha256='9f1a43e620530f9e570e4a493677d25f08310118d315b00e25a18f12913c41f0')
     version('7.5.1', sha256='e945f6e02854a74994c596d9db83444a1850c01648f1574adf144fbbabe05c97')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ipywidgets/package.py:18]: widgetsnbextension dep needs to be updated, ipython dep isn't correct:
```python
install_requires = setuptools_args['install_requires'] = [                               
    'ipykernel>=4.5.1',                                                                  
    'ipython_genutils~=0.2.0',                                                           
    'traitlets>=4.3.1',                                                                  
    # Requiring nbformat to specify bugfix version which is not required by              
    # notebook.                                                                          
    'nbformat>=4.2.0',                                                                   
    # TODO: Dynamically add this dependency                                              
    # only if notebook 4.x is installed in this                                          
    # interpreter, to allow ipywidgets to be                                             
    # installed on bare kernels.                                                         
    'widgetsnbextension~=3.6.0'                                                          
]                                                                                        
                                                                                         
extras_require = setuptools_args['extras_require'] = {                                   
    ':python_version<"3.3"' : ['ipython>=4.0.0,<6.0.0'],                                 
    ':python_version>="3.3"': ['ipython>=4.0.0'],                                        
    ':python_version>="3.6"': ['jupyterlab_widgets>=1.0.0'],                             
    'test:python_version=="2.7"': ['mock'],                                              
    'test': ['pytest>=3.6.0', 'pytest-cov'],                                             
}                         
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-ipywidgets/package.py [hunk 2]
```diff
@@ -37,4 +38,6 @@ class PyIpywidgets(PythonPackage):
     depends_on('py-widgetsnbextension@3.4.0:3.4', type=('build', 'run'),
                when='@7.4.2')
     depends_on('py-widgetsnbextension@3.5.0:3.5', type=('build', 'run'),
-               when='@7.5.1:')
+               when='@7.5.1:7.6.5')
+    depends_on('py-widgetsnbextension@3.6.0', type=('build', 'run'),
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ipywidgets/package.py]:
```suggestion
    depends_on('py-widgetsnbextension@3.6', type=('build', 'run'),
```
`3.6` and `3.6.0:3.6` are both equivalent to `~=3.6.0` used in the setup.py. Exactly `3.6.0` is too strict.

File: var/spack/repos/builtin/packages/py-ipywidgets/package.py [hunk 3]
```diff
@@ -37,4 +38,6 @@ class PyIpywidgets(PythonPackage):
...
+               when='@7.7.0')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ipywidgets/package.py]:
```suggestion
               when='@7.7:')
```
I think it's safe to assume this requirement will also exist in newer versions

File: var/spack/repos/builtin/packages/py-ipywidgets/package.py [hunk 4]
```diff
@@ -21,8 +22,8 @@ class PyIpywidgets(PythonPackage):
     depends_on('python@2.7:2.8,3.3:', type=('build', 'run'))
     # pip silently replaces distutils with setuptools
     depends_on('py-setuptools', type='build')
-    depends_on('py-ipython@4:', type=('build', 'run'))
-    depends_on('py-ipython@4:5', type=('build', 'run'), when='^python@:3.2')
+    depends_on('py-ipython@4:', type=('build', 'run'), when='^python@3.3:')
+    depends_on('py-ipython@4:5.8.0', type=('build', 'run'), when='^python@:3.3')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-ipywidgets/package.py]: Actually I think the previous ipython deps were correct, I'm not sure why I thought they were wrong. Sorry about that, can you revert that change?

Comment (aandvalenzuela) [reply var/spack/repos/builtin/packages/py-ipywidgets/package.py]: Sure, no problem! Thanks for the review

**Review by adamjstewart — DISMISSED** [review]

**Comment by adamjstewart** [comment]
You'll need to add a new version of `py-widgetsnbextension` to get the package audits to pass. Right now the version you added can't even be built. You can add it directly to this PR.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**